source/style-guide/markup/directives/sharedinclude.txt

```diff
@@ -0,0 +1,234 @@
+
+   The following table shows the compatibility between different versions of
+   the |driver-name| driver and the |driver-language|:
+
+The sourcing file could resemble the following code:
+
+.. code-block:: rst
+   :caption: compatibility-table.txt sourcing file in the documentation repository
+   :copyable: false
+
+  .. _sharedinclude:: drivers/compatibility-tables/language.rst
+
+     .. replacement:: driver-name
+
+        PyMongo
+
+     .. replacement:: driver-language
+
+        Python language
+
+The published page displays the following text:
+
+.. code-block:: rst
+   :copyable: false
+
+   The following table shows the compatibility between different versions of
+   the PyMongo driver and the Python language:
+
+Snooty reports a warning or error if the sourcing file is missing any
+replacement variables present in the external file.
+
+Placeholders in Inline Context
+``````````````````````````````
+
+An **inline context** describes the positioning of placeholders as adjacent to
+other paragraph text. You can only replace variable placeholders in an inline
+context with values that contain markup elements which includes any of the
+following types:
+
+- Unformatted text
+- Formatted text such as ``monospace``, **emphasis**, or *italic*
+- Links
+
+For example, suppose you want to source content from an external file, located
+at ``dbx/fundamentals/agg-operators.rst`` in the ``docs-shared`` repository,
+which contains the following content:
+
+.. code-block:: rst
+   :caption: dbx/fundamentals/agg-operators.rst external file in docs_shared
+   :copyable: false
+
+   To learn more about **aggregation operators**, see the
+   |learn-agg-operator-docs|.
+
+
+In the sourcing file, you can use the following placeholder replacement
+in the ``sharedinclude`` directive:
+
+.. code-block:: rst
+   :caption: agg-operators.txt sourcing file in a documentation repository
+   :copyable: false
+
+   .. sharedinclude:: dbx/fundamentals/agg-operators.rst
+
+      .. replacement::
+
+         Aggregation Operations page in the
+         :server:`MongoDB Server documentation </aggregation>`
+
+The published page renders as if the sourcing file contained the following rST:
+
+.. code-block:: rst
+   :copyable: false
+
+   To learn more about **aggregation operators**, see the Aggregation Operations
+   page in the :server:`MongoDB Server documentation </aggregation>`.
+
+
+You cannot replace variable placeholders in inline contexts with elements
+such as lists, codeblocks, includes, and headers. If you need to
+source these elements from a ``sharedinclude``,, make sure the placeholders
+are in a block context as described in the following section.
+
+Placeholders in Block Context
+`````````````````````````````
+
+A **block context** describes the positioning of placeholders as
+separated from paragraph text by linebreaks. You can replace variable
+placeholders in a block context with most inline or block rST elements
+including the following types:
```

Comment:
[Question]

I'm having a little trouble parsing this sentence. Is "including the following types" referring to either:

- The elements that I can replace variable placeholders with
- The types of rST elements where I can perform the replacement

Reply:
Updating this based on our discussion in Slack.